github/files/vagrant/puppet/Vagrantfile

```diff
@@ -36,11 +54,14 @@ Vagrant.configure("2") do |config|
     end
   end
 
-  data['vm']['synced_folder'].each do |i, folder|
-    if folder['source'] != '' && folder['target'] != '' && folder['id'] != ''
-      nfs = (folder['nfs'] == "true") ? "nfs" : nil
-      config.vm.synced_folder "#{folder['source']}", "#{folder['target']}", id: "#{folder['id']}", type: nfs
-    end
+  for folder in data['vm']['synced_folder'];
+    nfs = (folder['nfs'] == "true") ? "nfs" : nil
+    config.vm.synced_folder folder['source'], folder['target'],
+      id: folder['id'],
+      type: nfs,
+      rsync__auto: "true",
+      rsync__exclude: folder['excluded_paths'],
+      rsync__args: ["--verbose", "--archive", "--delete", "-z", "--chmod=ugo=rwX"]
```

Comment:
We are using nfs by default, but some windows machines don't support nfs, some people from our office prefer rsync method.